src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,120 @@
+import React from "react"
+import PropTypes from "prop-types"
+import styles from './Carrousel.module.scss'
+
+class CarouselElement extends React.Component {
+    constructor(props) {
+        super(props)
+        this.itemRef = React.createRef()
+    }
+
+
+    static propTypes = {
+        id: PropTypes.number.isRequired,
+        count: PropTypes.number.isRequired,
+        src: PropTypes.string.isRequired,
+        alt: PropTypes.string.isRequired,
+        text: PropTypes.array.isRequired,
+
+    }
+
+    componentDidMount() {
+        const state = (this.props.id === this.props.carouselCurrentSlideIndex)
+        this.itemRef.current.style.display = state
+            ? "block"
+            : "none"
+    }
+
+    componentDidUpdate() {
+        const state = (this.props.id === this.props.carouselCurrentSlideIndex)
+        this.itemRef.current.style.display = state
+            ? "block"
+            : "none"
+    }
+
+
+    render() {
+
+        return (
+            <div ref={this.itemRef}  >
+                <div className={`${styles.fade}`}>
+
+                    <img src={this.props.src} alt={this.props.alt} className={`${styles.styleCarrouselImage} ${this.props.className}`} />
+                    {this.props.text.map((element, index) => {
+                        return <div key={`${element.id} ${index}`} className={styles.text}>{element} </div>
+                    })}
+
+                    <div className={styles.text}>  ({this.props.id}/{this.props.count})</div>
+                </div>
+            </div>);
+
+    }
+
+};
+
+
+export default class Carousel extends React.Component {
+
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            carouselCurrentSlideIndex: props.carouselCurrentSlideIndex
+        };
+        this.prevClick = this.prevClick.bind(this);
```

Comment:
You can also use arrow function so you dont have to bind the function to the class in the constructor, is one of the simplifications of functional components. https://reactjs.org/docs/hooks-intro.html